sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -0,0 +1,305 @@
+          pipeline_spec.runtime_parameters[
+              arg.name].type = pipeline_spec_pb2.PrimitiveType.INT
+          pipeline_spec.runtime_parameters[
+              arg.name].default_value.int_value = arg.value
+        elif isinstance(arg.value, float):
+          pipeline_spec.runtime_parameters[
+              arg.name].type = pipeline_spec_pb2.PrimitiveType.DOUBLE
+          pipeline_spec.runtime_parameters[
+              arg.name].default_value.double_value = arg.value
+        elif isinstance(arg.value, str):
+          pipeline_spec.runtime_parameters[
+              arg.name].type = pipeline_spec_pb2.PrimitiveType.STRING
+          pipeline_spec.runtime_parameters[
+              arg.name].default_value.string_value = arg.value
+        else:
+          raise NotImplementedError(
+              'Unexpected parameter type with: "{}".'.format(str(arg.value)))
+
+    def _get_input_artifact_type(
+        input_name: str,
+        inputs: List[structures.InputSpec],
+    ) -> str:
+      """Find the input artifact type by input name.
+
+      Args:
+        input_name: The name of the component input.
+        inputs: The list of InputSpec
+
+      Returns:
+        The input type name if found in inputs, or '' if not found.
+      """
+      for component_input in inputs:
+        if component_input.name == input_name:
+          return type_utils.get_artifact_type(component_input.type)
+      return ''
+
+    deployment_config = pipeline_spec_pb2.PipelineDeploymentConfig()
+    importer_tasks = []
+
+    for op in pipeline.ops.values():
+      component_spec = op._metadata
+      task = pipeline_spec.tasks.add()
+      task.CopyFrom(op.task_spec)
+      deployment_config.executors[task.executor_label].container.CopyFrom(
+          op.container_spec)
+
+      # Check if need to insert importer node
+      for input_name in task.inputs.artifacts:
+        if not task.inputs.artifacts[input_name].producer_task:
+          artifact_type = _get_input_artifact_type(input_name,
+                                                   component_spec.inputs)
+
+          importer_task, importer_spec = importer_node.build_importer_spec(
+              task, input_name, artifact_type)
+          importer_tasks.append(importer_task)
+
+          task.inputs.artifacts[
+              input_name].producer_task = importer_task.task_info.name
+          task.inputs.artifacts[
+              input_name].output_artifact_key = importer_node.OUTPUT_KEY
+
+          deployment_config.executors[
+              importer_task.executor_label].importer.CopyFrom(importer_spec)
+
+    pipeline_spec.deployment_config.Pack(deployment_config)
+    pipeline_spec.tasks.extend(importer_tasks)
+
+    return pipeline_spec
+
+  def _create_pipeline(
+      self,
+      pipeline_func: Callable[..., Any],
+      pipeline_name: Optional[str] = None,
+      pipeline_description: Optional[str] = None,
+      params_list: List[dsl.PipelineParam] = None,
```

Comment:
Optional: `params_list` is introduced to workaround a problem when compiling TFX pipeline in `KubeflowDagRunner`. To simplify the v2 compiler we can remove this parameter and use `pipeline_func` as the SSoT